src/db/postgres.rs

```diff
@@ -39,4 +44,28 @@ impl Store for PostgresStore {
 
         Ok(row.0 == 0)
     }
+
+    async fn fetch_user_id<'a>(
+        &self,
+        user_id: &'a UserIdentifier,
+    ) -> Result<Option<Cow<'a, UserId>>, Box<dyn Error>> {
+        unimplemented!()
+    }
+
+    async fn fetch_password_hash(&self, user_id: &UserId) -> Result<super::PWHash, Box<dyn Error>> {
+        unimplemented!()
+    }
+
+    async fn check_otp(&self, user_id: &UserId, otp: &str) -> Result<bool, Box<dyn Error>> {
```

Comment:
We should settle on convention.  Either we use `check` or `is` for all boolean db options.  I am happy either way, but this and `is_username_available` should be consistent.

The other thing we should be clear on is that no business logic should exist in the storage layer.

Reply:
I imagined check_otp as just checking if the provided string exists in the db, if it really is a one time password. Probably need to do some actual research on what token auth in matrix means

Reply:
Therefore, since I imagined it as being able to be a single query, I figured it could be handled by the storage layer

Reply:
Ok.  Maybe `check_otp_exists`?

I am gonna make the same change to seeing if username is available `check_username_exists`.  This will make it more generic imo.

Reply:
sure sounds good